pkg/asset/machines/baremetal/machines.go

```diff
@@ -60,11 +62,18 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
-func provider(clusterName string, networkInterfaceAddress string, platform *baremetal.Platform, userDataSecret string) *baremetalprovider.BareMetalMachineProviderSpec {
+func provider(clusterName string, networkInterfaceAddress string, platform *baremetal.Platform, osImage string, userDataSecret string) *baremetalprovider.BareMetalMachineProviderSpec {
+	// The rhcos-downloader container launched by the baremetal-operator downloads the image,
+	// compresses it to speed up deployments and makes it available on provisioningIP via http
+	provisioningIP := platform.ClusterProvisioningIP
+	imageFilename := path.Base(osImage)
+	compressedImageFilename := strings.Replace(imageFilename, "openstack", "compressed", 1)
```

Comment:
can you add comment in terms of what API this is trying to follow??

Reply:
Ack sure, it's basically to align with the rhcos downloader filename convention:

https://github.com/openshift/ironic-rhcos-downloader/blob/master/get-resource.sh#L25

That downloads the openstack qcow2, then compresses it to make baremetal deployment faster (maybe we can figure out some way to do that by default in future).

I'll add an example of the expected URL and a link to the rhcos-downloader container as a comment.